goaviatrix/firewall_instance.go

```diff
@@ -119,6 +121,16 @@ func (c *Client) CreateFirewallInstance(firewallInstance *FirewallInstance) (str
 	if firewallInstance.UserData != "" {
 		addFirewallInstance.Add("user_data", firewallInstance.UserData)
 	}
+	if len(firewallInstance.Tags) > 0 {
+		tagList := make([]string, 0, len(firewallInstance.Tags))
+		for key, val := range firewallInstance.Tags {
+			escapedKey := strings.ReplaceAll(key, ":", "\\:")
+			escapedVal := strings.ReplaceAll(fmt.Sprint(val), ":", "\\:")
```

Comment:
Don't need fmt.Sprint

Reply:
firewallInstance.Tags is map[string]interface{}, so val is of type interface{} and needs to be cast to a string.